main.go

```diff
@@ -27,9 +28,11 @@ func main() {
 	}
 
 	// main message loop
-	var msg win.MSG
-	for win.GetMessage(&msg, 0, 0, 0) > 0 {
-		win.TranslateMessage(&msg)
-		win.DispatchMessage(&msg)
+	msg := (*win.MSG)(unsafe.Pointer(win.GlobalAlloc(0, unsafe.Sizeof(win.MSG{}))))
```

Comment:
For my own sake I'd like to better understand what this (and the following) line do. I'll start:

- `unsafe.Sizeof(win.MSG{})` ([reference](https://golang.org/pkg/unsafe/#Sizeof)) likely returns either 28 or 40 (depending on 32bit or 64bit OS) See https://pkg.go.dev/github.com/lxn/win#MSG
- that value is used in [`win.GlobalAlloc`](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-globalalloc) to allocate a small chunk of bytes on the heap. (aside, I'm not worried about the overhead warnings on MSDN)
- `unsafe.Pointer` is making an unsafe pointer to that allocated heap memory
- `(*win.MSG)` is casting all of that into a pointer for a `win.MSG`.

Looking in the docs, `win.GlobalAlloc` returns a `HGLOBAL`, I'll suggest:

```suggestion
	hglobal := win.GlobalAlloc(0, unsafe.Sizeof(win.MSG{}))
	msg := (*win.MSG)(unsafe.Pointer(hglobal))
```